### PR TITLE
Update model interface for optimal filter

### PR DESCRIPTION
**Pull request by tkoskela**
As planned in #136 

Split `update_particles!` to `update_particle_dynamics!` and `update_particle_noise!`

DONE: Now both new functions return the observations which may not be ideal. Would it be better to separate `get_particle_observations` into a third function?

### CODE REVIEW

**Comment by giordano** [comment]
I pushed this last week https://github.com/Team-RADDISH/ParticleDA.jl/commit/28feb5234a87e8c3514e6a4ea741758bbc77cbe2 but forgot to open the PR :facepalm:  Yours is better, it has also tests! :smiley:

**Comment by tkoskela** [comment]
That's exactly the thing in my TODO! Can we merge it into this PR?

**Comment by tkoskela** [comment]
Yes we can! 💪

File: docs/src/index.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ be used by [`run_particle_filter`](@ref):
 ParticleDA.get_particles
 ParticleDA.get_truth
 ParticleDA.update_truth!
+ParticleDA.get_observations!
 ParticleDA.update_particles!
```
Comment (giordano) [inline docs/src/index.md:72]: I think you need to do
```suggestion
ParticleDA.update_particle_dynamics!
ParticleDA.update_particle_noise!
ParticleDA.get_particle_observations!
```

**Comment by tkoskela** [comment]
I'll mark this as ready for review. The other thing to do in #136 is expose more fields from the model_data struct but we can do that in another PR.

**Review by giordano — APPROVED** [review]

**Pull request merged**